src/services/impl/GameContext.hh

```diff
@@ -0,0 +1,32 @@
+#ifndef RA_SERVICES_GAMECONTEXT_HH
+#define RA_SERVICES_GAMECONTEXT_HH
+#pragma once
+
+#include "services\IGameContext.hh"
+
+namespace ra {
+namespace services {
+namespace impl {
+
+class GameContext : public IGameContext
+{
+public:
+    void LoadGame(unsigned int nGameId, const std::wstring& sGameTitle) override;
+    
+    unsigned int GameId() const override { return m_nGameId; }       
+    const std::wstring& GameTitle() const override { return m_sGameTitle; }
+
+    const std::string& GameHash() const override { return m_sGameHash; }
+    void SetGameHash(const std::string& sGameHash) override { m_sGameHash = sGameHash; }
+
+private:
+    unsigned int m_nGameId = 0;
+    std::wstring m_sGameTitle;
+    std::string m_sGameHash;
```

Comment:
Just my opinion, not imperative. There are many ways to about this.

If you're going to have getters and setters for every field you might as well make this a struct. Another thing you could do is till keep it a class and only friend classes/functions that need access.

Reply:
Not every field, most of the fields should be set by the data returned from the API call. 

Long term, I expect one input (the hash), which will make a server call to get the ID, which will make another server call to get the rest of the data, which will include the achievements - I plan to move g_pCoreAchievements, g_pUnofficialAchievements, and g_pLocalAchievements into this class eventually. I'll probably end up moving the code notes into here too at some point.

The hashing resolution might end up separated due to the logic around popping the dialog when a hash doesn't resolve. I'm still trying to decide the best way to handle that.